scylla/src/observability/metrics.rs

```diff
@@ -218,7 +218,7 @@ pub struct Metrics {
 }
 
 impl Metrics {
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
         Metrics::default()
     }
```

Comment:
> As the user should not be able to create metrics instance otherwise
> than by `get_metrics()` function, the `Metrics::new()` method shall
> be set to pub(crate) visibility to support only internal usage.

In order to fullfill the goal stated in the commit message, you need to make `impl Default for Metrics` guarded by `cfg(test)`, otherwise user can still create an instance of this struct.